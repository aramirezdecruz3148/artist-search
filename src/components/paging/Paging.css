@import url('https://fonts.googleapis.com/css?family=Monoton|Poiret+One|Viga&display=swap');

.paging {
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Viga', sans-serif;
  color: rgb(66, 65, 65);
  font-size: 1.3vw;
}