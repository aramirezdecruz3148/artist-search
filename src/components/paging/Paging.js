import React from 'react';
import PropTypes from 'prop-types';
import styles from './Paging.css';

function Paging({ 
  onClickPrevious, 
  onClickNext, 
  disableNext, 
  disablePrev,
  currentPage,
  totalPages 
}) {
  return (
    <div className={styles.paging}>
      <button onClick={onClickPrevious} disabled={disablePrev}>⇦</button>
      <pre>Page {currentPage} of {totalPages}</pre>
      <button onClick={onClickNext} disabled={disableNext}>⇨</button>
    </div>
  );
}

Paging.propTypes = {
  onClickPrevious: PropTypes.func.isRequired,
  onClickNext: PropTypes.func.isRequired,
  disableNext: PropTypes.any.isRequired,
  disablePrev: PropTypes.any.isRequired,
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired
};

export default Paging;
