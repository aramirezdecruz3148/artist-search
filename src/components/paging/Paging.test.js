import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging onClickPrevious={() => {}} onClickNext={() => {}} disableNext={9} disablePrev={9} currentPage={10} totalPages={3}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
