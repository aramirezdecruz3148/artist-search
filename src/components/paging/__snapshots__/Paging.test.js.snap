// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<div
  className="paging"
>
  <button
    disabled={9}
    onClick={[Function]}
  >
    Prev
  </button>
  <p>
    Page 
    10
     of 
    3
  </p>
  <button
    disabled={9}
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
